backend/multimodalexplorer/functions/helpers.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+import os
+
+import pandas as pd
+
+
+def save_data_as_file(data, file_name, folder="../db"):
+    folder_path = os.path.join(os.path.dirname(__file__), folder)
+    os.makedirs(folder_path, exist_ok=True)
+    csv_path = os.path.join(folder_path, f"{file_name}.tsv")
+
+    df = pd.DataFrame(data)
+
+    df.to_csv(csv_path, sep="\t", index=False)
```

Comment:
not totally sure why you go back to pandas to save to file. The numpy embeddings should saved with numpy.save/load. For umap, you get a bunch of 2 dimentional points, you could also just dump this with numpy. I see that you try to keep more columns with dataset names etc. In this case saving to CSV might be useful.
For faiss index, you should not save it as a dataframe, faiss has a save API: ` faiss.write_index(returned_index, index_output_file)` and ` faiss.read_index(filename)`.

Reply:
ok, I looked at the umap API a bit more in details, `fit_transform` returns the whole array of points. If you want to save this as csv in a useful way, you will have to unflatten the points probably.